lib/src/kbasesearchengine/parse/ObjectParser.java

```diff
@@ -80,7 +121,7 @@ public static String extractParentFragment(ObjectTypeParsingRules parsingRules,
             if (!rules.isFromParent()) {
                 continue;
             }
-            //TODO CODE this seems wrong. Why adding null paths?
+
```

Comment:
Could you restore this TODO? Adding nulls to this list still seems like the wrong thing to do to me, and if it ever actually happens I expect NPEs to get thrown.

Reply:
Done. I'll fix this and test in the next PR.